OuaReadmeUpdate (#330)
IBM/universal-connectors: filter-plugin/logstash-filter-oua-guardium/README.md

1. Unified auditing must be enabled in an Oracle database that will be monitored by this method
2. Download the Basic Instant client package from Oracle.

>    **Note:** : In this release,only specific instance client will be supported from starting v21.1.0.0.0 Download [here](https://download.oracle.com/otn_software/linux/instantclient/211000/oracle-instantclient-basic-21.1.0.0.0-1.x86_64.rpm)

Reviewer: ```suggestion
   **Note:** : In this release,only specific instance clients will be supported starting the v21.1.0.0.0 Download [here](https://download.oracle.com/otn_software/linux/instantclient/211000/oracle-instantclient-basic-21.1.0.0.0-1.x86_64.rpm)
```

3. Download the OUA universal connector plug-in  `guardium-oua-uc.zip` from [here](https://github.com/IBM/universal-connectors/raw/main/filter-plugin/logstash-filter-oua-guardium/OracleUnifiedAuditPackage/OracleUnifiedAudit/guardium-oua-uc.zip)
4. A designated user for OUA UC should be created for Oracle database access. An existing user with sysdba privileges can also be used
5. You must create a secret containing your OUA universal connector password.

>     - Example: `grdapi universal_connector_keystore_add key=OUA_USER_PASS password=<PASSWORD>` where `<PASSWORD>` is the OUA universal connector user’s password for the database. `OUA_USER_PASS` will be used in the plug-in configuration as a variable for the password secret

Reviewer: ```suggestion
    - Example: `grdapi universal_connector_keystore_add key=OUA_USER_PASS password=<PASSWORD>` where `<PASSWORD>` is the OUA universal connector user’s password for the database. `OUA_USER_PASS` will be used in the plug-in configuration as a variable for the password secret.
```

4. Click on the "UPLOAD” button and upload downloaded `oracle-instantclient-basic-21.1.0.0.0-1.x86_64.rpm` and then `guardium-oua-uc.zip`

>    5. Click on the "+". The Connector Configuration dialog opens 

Reviewer: ```suggestion
   5. Click on the "+". The Connector Configuration dialog opens.
```

>        - Type any unique name in the **Connector name** field

Reviewer: ```suggestion
       - Type any unique name in the **Connector name** field.
```

                             add_field => {"SERVER_PORT" => "<Enter_Server_Port>"} }
           ```

>          - Paste content of "[ouaPipe.conf](https://github.com/IBM/universal-connectors/blob/main/filter-plugin/logstash-filter-oua-guardium/ouaPipe.conf)" in the **Filter configuration** field

Reviewer: ```suggestion
         - Paste the content from the "[ouaPipe.conf](https://github.com/IBM/universal-connectors/blob/main/filter-plugin/logstash-filter-oua-guardium/ouaPipe.conf)" file in the **Filter configuration** field.
```

             ```
                if [type] == "oua" {
             json {
                 source => "message"
             }
                 mutate {
                         add_field => {"[HostName]" => "%{SERVER_ADDRESS}" }
                         add_field => {"[PortNumber]" => "%{SERVER_PORT}" }
                         }
             if "_jsonparsefailure" not in [tags] {
                 oua_filter {}
             }
           }

             ```

>        - **NOTE**: The type specified for the filters must be unique among Universal Connectors,and be identical in the input and filter configurations

Reviewer: ```suggestion
       - **NOTE**: The type specified for the filters must be unique among universal connectors,and be identical in the input and filter configurations
```

## Limitation

> Normally, the "statement Type" attribute for the "FULL SQL" entity in Guardium reports shows us whether a full SQL statement is a prepared statement. However, because OUA doesn't give us information about whether a statement is a prepared statement or not, the "Statement type" attribute is not applicable for the OUA universal connector plug-in.

Reviewer: ```suggestion
Normally, the "statement type" attribute for the "FULL SQL" entity in Guardium reports shows us whether a full SQL statement is a prepared statement. However, because OUA doesn't give us information about whether a statement is a prepared statement or not, the "statement type" attribute is not applicable for the OUA universal connector plug-in.
```
